pkg/cmd/admin/router/router.go

```diff
@@ -288,15 +298,45 @@ func generateSecretsConfig(cfg *RouterConfig, kClient *kclient.Client,
 				},
 			},
 		}
+		volumes = append(volumes, volume)
 
-		secretsMount := kapi.VolumeMount{
+		mount := kapi.VolumeMount{
 			Name:      secretsVolumeName,
 			ReadOnly:  true,
 			MountPath: secretsPath,
 		}
+		mounts = append(mounts, mount)
+	}
+
+	if len(defaultCert) > 0 {
+		// TODO: extract the private key from the CRT and set it as its own key
+		// TODO: use upstream constants for this
+		secret := &kapi.Secret{
+			ObjectMeta: kapi.ObjectMeta{
+				Name: fmt.Sprintf("%s-certs", cfg.Name),
+			},
+			Type: "kubernetes.io/tls",
```

Comment:
Not in our branch yet

On Wed, Feb 17, 2016 at 11:51 AM, Michail Kargakis <notifications@github.com
> wrote:

> In pkg/cmd/admin/router/router.go
> <https://github.com/openshift/origin/pull/7251#discussion_r53193982>:
>
> >  			Name:      secretsVolumeName,
> >  			ReadOnly:  true,
> >  			MountPath: secretsPath,
> >  		}
> > +		mounts = append(mounts, mount)
> > +	}
> > +
> > +	if len(defaultCert) > 0 {
> > +		// TODO: extract the private key from the CRT and set it as its own key
> > +		// TODO: use upstream constants for this
> > +		secret := &kapi.Secret{
> > +			ObjectMeta: kapi.ObjectMeta{
> > +				Name: fmt.Sprintf("%s-certs", cfg.Name),
> > +			},
> > +			Type: "kubernetes.io/tls",
>
>
> https://github.com/kubernetes/kubernetes/blob/0866c377e9a84a73e0ff78efdd1def7f7e8b437a/pkg/api/types.go#L2311
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/7251/files#r53193982>.
>